tv_apps/remote-control/js/app.js

```diff
@@ -0,0 +1,144 @@
+/* global KeyNavigationAdapter, Section, Config */
+'use strict';
+
+(function(exports) {
+  var MAIN_SECTION = 'main-section';
+  var DEFAULT_ELEMENT = 'qrcode';
+
+  function App() {
+    this.config = null;
+    this.sections = [];
+    this.keyNav = null;
+    this.ip = '';
+  }
+
+  App.prototype = {
+    start: function() {
+      var sections = document.getElementsByTagName('section');
+      Array.from(sections).forEach((dom) => {
+        var id = dom.id;
+        var section = new Section(id);
+        if (id == MAIN_SECTION) {
+          this.activeSection = section.show();
+        } else {
+          section.hide();
+        }
+        this.sections[id] = section;
+      });
+
+      this.keyNav = new KeyNavigationAdapter();
+      this.keyNav.init();
+      this.keyNav.on('move', (direction) => {
+        this.activeSection.move(direction);
+      });
+      this.keyNav.on('enter', this._handleClick.bind(this));
+      this.keyNav.on('esc', this._handleBack.bind(this));
+
+      this.config = new Config();
+      this.config.start();
+      this.config.once('ready', this._statusCheck.bind(this));
+      this.config.on('changed', this._onSettingsChanged.bind(this));
+    },
+
+    stop: function() {
+      this.config.stop();
+      this.config = null;
+
+      this.keyNav.uninit();
+      this.keyNav = null;
+
+      this.activeSection = '';
+      this.sections = [];
+    },
+
+    _statusCheck: function() {
```

Comment:
I didn't see too many "check" action but a few status reset, we should rename this function to be something meaningful and add comments to describe its purpose.